pkg/columns/filter/filter.go

```diff
@@ -263,6 +266,40 @@ func getComparisonFuncForComparisonType[OT constraints.Ordered, T any](ct compar
 	}
 }
 
+func getComparisonFuncAsStringField[T any](ct comparisonType, negate bool, column *columns.Column[T], refValue any, re *regexp.Regexp) func(a *T) bool {
+	ff := columns.GetFieldAsString[T](column)
+	switch ct {
+	case comparisonTypeRegex:
+		return func(a *T) bool {
+			return re.MatchString(ff(a)) != negate
+		}
+	case comparisonTypeMatch:
+		return func(a *T) bool {
+			return ff(a) == refValue.(string) != negate
+		}
+	case comparisonTypeGt:
+		return func(a *T) bool {
+			return ff(a) > refValue.(string) != negate
+		}
+	case comparisonTypeGte:
+		return func(a *T) bool {
+			return ff(a) >= refValue.(string) != negate
+		}
+	case comparisonTypeLt:
+		return func(a *T) bool {
+			return ff(a) < refValue.(string) != negate
+		}
+	case comparisonTypeLte:
+		return func(a *T) bool {
+			return ff(a) <= refValue.(string) != negate
+		}
+	default:
+		return func(a *T) bool {
+			return false
+		}
+	}
+}
+
```

Comment:
What about refactor `getComparisonFuncForComparisonType` to receive `ff`, then reuse it in `case reflect.Array, reflect.Slice:` instead of duplicating the logic on this new `getComparisonFuncAsStringField`. 

For sure, easier to understand with some code: https://github.com/inspektor-gadget/inspektor-gadget/commit/887d62873ed798ea4c8b5684b23258ad2fc7edf6